ethereum/PFP.sol

```diff
@@ -172,8 +182,10 @@ contract PFP is ERC721A, Ownable, VRFConsumerBase {
     // this is callback, it will be called by the vrf coordinator
     function fulfillRandomness(bytes32 requestId, uint256 randomness) internal override {
         request_id = requestId;
-        if(revealed && randStartPos==0) {
+        if(randStartPos==0) {
```

Comment:
Changed the flow here. Previously the reveal function call did this:
* Made the call to VRF for the random number
* Set the collection to revealed
At this point we are waiting for VRF to respond with the random number to set the start position. So during this time we are a) revealed, and b) don't have the random start position. gulp. This should only be a handful of blocks, but if gas is crazy could be longer.

In the very worst case scenario people start revealing metadata that doesn't have the random offset (so it's just in order), and getting excited and/or trying to flip items (that will change again when the VRF call gets back...).

With my suggested order the `reveal()` function just make the VRF call. At this point we are still pre-reveal. When the VRF call gets back (here) it sets the random start position and sets the collection to revealed at the same time, so there is no gap where we have a revealed collection but an un-randomised metadata start position.